File: src/tests/misc-c/getuid.c
/*
 * Copyright(c) The Maintainers of Nanvix.
 * Licensed under the MIT License.
 */

//==================================================================================================
// Imports
//==================================================================================================

#include "common.h"
#include <assert.h>
#include <stdio.h>
#include <unistd.h>

//==================================================================================================
// Standalone Functions
//==================================================================================================

// Tests whether we can the user ID of the calling process with `getuid()`.
void test_getuid(void)
{
    fprintf(stderr, "testing getuid() ... ");

    assert(getuid() != (uid_t)-1);

    fprintf(stderr, "passed\n");
}

File: src/libs/syscall/src/unistd/syscall/getuid.rs
// Copyright(c) The Maintainers of Nanvix.
// Licensed under the MIT License.

//==================================================================================================
// Imports
//==================================================================================================

use ::sys::error::Error;
use ::sysapi::sys_types::uid_t;

//==================================================================================================
// Standalone Functions
//==================================================================================================

///
/// # Description
///
/// Returns the user ID of the calling process.
///
/// # Returns
///
/// Upon successful completion, `getuid()` returns the user ID of the calling process. Otherwise, it
/// returns an error.
///
pub fn getuid() -> Result<uid_t, Error> {
    ::syslog::trace!("getuid()");
    Ok(usize::from(::sys::pm::UserIdentifier::ROOT) as uid_t)
}
